SConstruct

```diff
@@ -109,6 +110,7 @@ def set_defaults(env):
     env.Append(CCFLAGS=['-g', '-Wshadow', '-Wall', '-Wno-missing-braces',
                         '-fpic', '-D_GNU_SOURCE', '-DD_LOG_V2'])
     env.Append(CCFLAGS=['-O2', '-DDAOS_VERSION=\\"' + DAOS_VERSION + '\\"'])
+    env.Append(CCFLAGS=['-DAPI_VERSION=\\"' + API_VERSION + '\\"'])
```

Comment:
Is there somewhere we use this in the code?   should we consider adding API version to the internal prints as well?

Reply:
I don't think we need it as globally as `DAOS_VERSION`.  We might not even need it anywhere else.  I just figured I'd add it here for any potential future use?  Want to defer that until it's needed?

Reply:
I was thinking something like daos version 1.0.0, api version 0.9.5  or some such.   We can defer

Reply:
But this really only applies in client contexts, right?

Reply:
Correct.   Does DFS need something similar?   I presume we could also have divergence between libdfs  and libioil and the main libdaos library version?

Reply:
I suppose we could.  Maybe it's sufficient to version all of the client APIs together so that they all use the same version.